File: completed-labs/18/java/TrieQueryTest.java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;
import java.util.Random;
import org.junit.Test;

public class TrieQueryTest {
    @Test
    public void testQueryOne() {
	Trie t = new Trie();
	t.insert("one");
	assertTrue(!t.query("a"));
	assertTrue(!t.query("on"));
	assertTrue(t.query("one"));
	assertTrue(!t.query("ones"));
	assertTrue(!t.query("two"));
	assertTrue(!t.query("only"));
    }
    @Test
    public void testQueryTwo() {
	Trie t = new Trie();
	t.insert("one");
	t.insert("two");
	assertTrue(!t.query("a"));
	assertTrue(!t.query("on"));
	assertTrue(t.query("one"));
	assertTrue(!t.query("ones"));
	assertTrue(!t.query("t"));
	assertTrue(!t.query("tw"));
	assertTrue(t.query("two"));
	assertTrue(!t.query("twos"));
	assertTrue(!t.query("only"));
    }
    @Test
    public void testQueryCommonPrefix() {
	Trie t = new Trie();
	t.insert("one");
	t.insert("only");
	assertTrue(!t.query("a"));
	assertTrue(!t.query("on"));
	assertTrue(t.query("one"));
	assertTrue(!t.query("ones"));
	assertTrue(!t.query("t"));
	assertTrue(!t.query("tw"));
	assertTrue(!t.query("two"));
	assertTrue(!t.query("twos"));
	assertTrue(t.query("only"));
    }
    @Test
    public void testQueryStock() {
	Trie t = new Trie();
	assertTrue(!t.query("bear"));
	t.insert("bear");
	assertTrue(t.query("bear"));
	assertTrue(!t.query("bell"));
	t.insert("bell");
	assertTrue(t.query("bell"));
	assertTrue(!t.query("bid"));
	t.insert("bid");
	assertTrue(t.query("bid"));
	assertTrue(!t.query("bull"));
	t.insert("bull");
	assertTrue(t.query("bull"));
	assertTrue(!t.query("buy"));
	t.insert("buy");
	assertTrue(t.query("buy"));
	assertTrue(!t.query("sell"));
	t.insert("sell");
	assertTrue(t.query("sell"));
	assertTrue(!t.query("stock"));
	t.insert("stock");
	assertTrue(t.query("stock"));
	assertTrue(!t.query("stop"));
	t.insert("stop");
	assertTrue(t.query("bear"));
	assertTrue(t.query("bell"));	
	assertTrue(t.query("bid"));
	assertTrue(t.query("bull"));	
	assertTrue(t.query("buy"));
	assertTrue(t.query("sell"));
	assertTrue(t.query("stock"));
	assertTrue(t.query("stop"));
    }
}

File: completed-labs/18/cpp/TrieQueryTest.cpp
#include <cppunit/extensions/HelperMacros.h>
#include <cppunit/CompilerOutputter.h>
#include <cppunit/extensions/TestFactoryRegistry.h>
#include <cppunit/ui/text/TestRunner.h>

#include "Trie.hpp"

class TrieQueryTest : public CppUnit::TestFixture {
  CPPUNIT_TEST_SUITE(TrieQueryTest);
  CPPUNIT_TEST(testQueryOne);
  CPPUNIT_TEST(testQueryTwo);
  CPPUNIT_TEST(testQueryCommonPrefix);
  CPPUNIT_TEST(testQueryStock);
  CPPUNIT_TEST_SUITE_END();
public:
  void setUp() { };
  void tearDown() { };

  void testQueryOne();
  void testQueryTwo();
  void testQueryCommonPrefix();
  void testQueryStock();
};

void TrieQueryTest::testQueryOne() {
  Trie *t = new Trie();
  t->insert("one");
  CPPUNIT_ASSERT(!t->query("a"));
  CPPUNIT_ASSERT(!t->query("on"));
  CPPUNIT_ASSERT(t->query("one"));
  CPPUNIT_ASSERT(!t->query("ones"));
  CPPUNIT_ASSERT(!t->query("two"));
  CPPUNIT_ASSERT(!t->query("only"));
}

void TrieQueryTest::testQueryTwo() {
  Trie *t = new Trie();
  t->insert("one");
  t->insert("two");
  CPPUNIT_ASSERT(!t->query("a"));
  CPPUNIT_ASSERT(!t->query("on"));
  CPPUNIT_ASSERT(t->query("one"));
  CPPUNIT_ASSERT(!t->query("ones"));
  CPPUNIT_ASSERT(!t->query("t"));
  CPPUNIT_ASSERT(!t->query("tw"));
  CPPUNIT_ASSERT(t->query("two"));
  CPPUNIT_ASSERT(!t->query("twos"));
  CPPUNIT_ASSERT(!t->query("only"));
}

void TrieQueryTest::testQueryCommonPrefix() {
  Trie *t = new Trie();
  t->insert("one");
  t->insert("only");
  CPPUNIT_ASSERT(!t->query("a"));
  CPPUNIT_ASSERT(!t->query("on"));
  CPPUNIT_ASSERT(t->query("one"));
  CPPUNIT_ASSERT(!t->query("ones"));
  CPPUNIT_ASSERT(!t->query("t"));
  CPPUNIT_ASSERT(!t->query("tw"));
  CPPUNIT_ASSERT(!t->query("two"));
  CPPUNIT_ASSERT(!t->query("twos"));
  CPPUNIT_ASSERT(t->query("only"));
}

void TrieQueryTest::testQueryStock() {
  Trie *t = new Trie();
  CPPUNIT_ASSERT(!t->query("bear"));
  t->insert("bear");
  CPPUNIT_ASSERT(t->query("bear"));
  CPPUNIT_ASSERT(!t->query("bell"));
  t->insert("bell");
  CPPUNIT_ASSERT(t->query("bell"));
  CPPUNIT_ASSERT(!t->query("bid"));
  t->insert("bid");
  CPPUNIT_ASSERT(t->query("bid"));
  CPPUNIT_ASSERT(!t->query("bull"));
  t->insert("bull");
  CPPUNIT_ASSERT(t->query("bull"));
  CPPUNIT_ASSERT(!t->query("buy"));
  t->insert("buy");
  CPPUNIT_ASSERT(t->query("buy"));
  CPPUNIT_ASSERT(!t->query("sell"));
  t->insert("sell");
  CPPUNIT_ASSERT(t->query("sell"));
  CPPUNIT_ASSERT(!t->query("stock"));
  t->insert("stock");
  CPPUNIT_ASSERT(t->query("stock"));
  CPPUNIT_ASSERT(!t->query("stop"));
  t->insert("stop");
  CPPUNIT_ASSERT(t->query("bear"));
  CPPUNIT_ASSERT(t->query("bell"));	
  CPPUNIT_ASSERT(t->query("bid"));
  CPPUNIT_ASSERT(t->query("bull"));	
  CPPUNIT_ASSERT(t->query("buy"));
  CPPUNIT_ASSERT(t->query("sell"));
  CPPUNIT_ASSERT(t->query("stock"));
  CPPUNIT_ASSERT(t->query("stop"));
}

CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(TrieQueryTest, "Query");
CPPUNIT_REGISTRY_ADD_TO_DEFAULT("Query");
